app/services/ProxyFrameworkQueue.scala

```diff
@@ -253,19 +271,34 @@ class ProxyFrameworkQueue @Inject() (config: Configuration,
         val updatedScanTarget = scanTarget.copy(pendingJobIds = updatedPendingJobIds, transcoderCheck = Some(tcReport))
         scanTargetDAO.put(updatedScanTarget)
 
-        val updatedJobDesc = jobDesc.copy(jobStatus = JobStatus.ST_SUCCESS, completedAt = Some(ZonedDateTime.now()))
-        jobModelDAO.putJob(updatedJobDesc)
+        val updatedJobDesc = jobDesc.copy(jobStatus = actualStatus, completedAt = Some(ZonedDateTime.now()))
+        jobModelDAO.putJob(updatedJobDesc).onComplete({
+          case Success(Some(Left(err)))=>
+            logger.error(s"Could not update job ${jobDesc.jobId}: $err")
+          case Success(Some(Right(rec)))=>
+            logger.info(s"Updated job ${jobDesc.jobId}")
+            sqsClient.deleteMessage(new DeleteMessageRequest().withQueueUrl(rq.getQueueUrl).withReceiptHandle(receiptHandle))
+          case Success(None)=>
+            logger.info(s"Updated job ${jobDesc.jobId}")
+            sqsClient.deleteMessage(new DeleteMessageRequest().withQueueUrl(rq.getQueueUrl).withReceiptHandle(receiptHandle))
+          case Failure(err)=>
+            logger.error(s"update thread failed: ", err)
+        })
 
-        if(jobDesc.jobStatus==JobStatus.ST_ERROR){
+        if(actualStatus==JobStatus.ST_ERROR){
           ownRef ! HandleFailure(msg, jobDesc, rq, receiptHandle, originalSender)
         } else {
-          sqsClient.deleteMessage(new DeleteMessageRequest().withQueueUrl(rq.getQueueUrl).withReceiptHandle(receiptHandle))
```

Comment:
this is now done conditionally on whether the job update succeeded. If the job update failed, it gets retried.